.vscode/settings.json

```diff
@@ -0,0 +1,9 @@
+{
+  "editor.formatOnSave": true,
+  "editor.codeActionsOnSave": {
+    "source.fixAll": true
+  },
+  "[typescriptreact]": {
```

Comment:
What about Javascript? Does it use `prettier-eslint` by default?

Reply:
Honestly might be better to use prettier ignore files if you want to exclude something

Reply:
This is how VSCode set it up for me so I copied this from my local config and shared it here.

Reply:
Want to confirm it works for `.ts, .tsx, .js, .jsx` and runs only prettier for the rest

Reply:
I think `typescripreact` only means `.ts, .tsx`. When I was trying this it was working fine.

Reply:
So can I presumably just add another block for [javascriptreact] ?

Reply:
yes, but do we need it ? because this is a typescript project and we won't be using `.js,.jsx` extensions

Reply:
We need it for `.js` and `.jsx` because there may be cases where it is not possible to use TS (index.js is an example of this), plus, I will be using this as the base for our shared linting config

Reply:
I've set the prettier to be the default formatter for the workspace and added a block for javascript.

In addition I also added emmet.

Reply:
What about plain Javascript and TS? I think `javascriptreact` must refer to `.jsx` only, unless I'm mistake?

Reply:
Ok I'll add those too lol